src/main/java/Course.java

```diff
@@ -191,6 +192,8 @@ public boolean equals(Object other) {
         && o.year == this.year && o.semester.equals(semester);
     }
 
+    public boolean has_no_times() {return times == null || times.isEmpty();}
```

Comment:
So this returns two separate assertions? Where do these get called specifically? Is "times == null" for  single courses and "times.isEmpty()" for checking a course while in search?